### PR TITLE
added support for cloning a document

### PR DESCRIPTION
**Pull request by kunal15595**

### CODE REVIEW

File: src/components/ModalComponent.tsx [hunk 1]
```diff
@@ -157,11 +157,19 @@ export class ModalComponent extends React.Component<ModalComponentProps, any> {
                     </Modal.Footer>
                 </Modal>
                 }
-                <div className="btn btn-primary"
-                     onClick={this.modalPerformOperation(this.props.modalType)}>{this.props.modalType === "EDIT" ? "Save" : this.props.modalType === "CREATE" ? "Create" : "Filter"}</div>
+                {this.props.modalType === "EDIT" ?
+                    <>
+                        <div className="btn btn-primary" onClick={this.modalPerformOperation(this.props.modalType, true)}>Update</div>
+                        <div className="btn btn-primary" onClick={this.modalPerformOperation(this.props.modalType, false)}>Save as New</div>
```
Comment (kunal15595) [inline src/components/ModalComponent.tsx]: Two options in editModal - `Save as New` and `Update`

**Comment by kunal15595** [comment]
@ankitcf yes, for now it'll create issues, client side is not the correct place to remove `_id`, so it requires changes on server-side.
I'm not sure what is the best way to fix this, server-side db level services or something very low as `mongo-utils`

**Comment by ankitcf** [comment]
Agreed. For now, the controller should remove it. Lets not change mongo
utils because lot of objects use some other field instead of _id as primary
key which mongo util might not be aware of.

**Comment by kunal15595** [comment]
cool

**Pull request merged**